aim/web/ui/src/modules/core/engine/blob-uri-system/index.ts

```diff
@@ -0,0 +1,159 @@
+import _ from 'lodash-es';
+
+import { createBlobsRequest } from 'modules/core/api/runsApi';
+
+import { throttle } from 'components/Table/utils';
+
+import { SequenceTypesEnum } from 'types/core/enums';
+
+import { parseStream } from 'utils/encoder/streamEncoding';
+import arrayBufferToBase64 from 'utils/arrayBufferToBase64';
+
+type Callback = (blobData: string) => void;
+
+export interface IBlobURISystemEngine {
+  engine: {
+    fire: (blobUri: string, blobData: any) => void;
+    on: (blobUri: string, callback: Callback) => () => void;
+    unsubscribe: (blobUri: string, callback: Callback) => void;
+    getBlobData: (blobUri: string) => string | null;
+    addUriToQueue: (blobUri: string) => void;
+    getBlobsData: (uris: string[]) => any;
+  };
+}
+
+const BATCH_SEND_DELAY = 1000;
+
+function createBlobURISystemEngine(
+  sequenceType: SequenceTypesEnum,
+): IBlobURISystemEngine {
+  let blobsData: Record<string, string> = {};
```

Comment:
Why those bindings are using let ?

Reply:
Changed the blobsData and blobsSubscriptions to const.